src/scripts/clipperUI/clipperStateUtilities.ts

```diff
@@ -0,0 +1,46 @@
+import {StringUtils} from "../stringUtils";
+
+import {ClipMode} from "./clipMode";
+import {ClipperState} from "./clipperState";
+import {Status} from "./status";
+
+export module ClipperStateUtilities {
+	export function isUserLoggedIn(state: ClipperState): boolean {
+		return (state.userResult && state.userResult.status && state.userResult.data && !!state.userResult.data.user);
+	}
+
+	export function clipButtonEnabled(clipperState: ClipperState): boolean {
+		let currentMode = clipperState.currentMode.get();
+		switch (currentMode) {
+			case ClipMode.Pdf:
+				if (!clipperState.pdfPreviewInfo.isLocalFileAndNotAllowed) {
+					return false;
+				} else if (clipperState.pdfResult.status !== Status.Succeeded) {
+					return false;
+				} else if (clipperState.pdfPreviewInfo.allPages) {
+					return true;
+				}
+				// We know at this point that the status is succeeded and the user has a specific page range, so if
+				// it is unparseable or empty, we disallow the clip
+				let pages = StringUtils.parsePageRange(clipperState.pdfPreviewInfo.selectedPageRange);
```

Comment:
What about the extra parameter which passes the total number of pages?  Without it we won't be able to validate the upper page limit, no?